django/conf/__init__.py

```diff
@@ -180,6 +180,9 @@ def __init__(self, settings_module):
             os.environ['TZ'] = self.TIME_ZONE
             time.tzset()
 
+    def __dir__(self):
```

Comment:
I reverted this change and didn't see any test failures.

Reply:
I've added a test for it in #11012 but it'll only fail before 573f44d.